src/Visualization/Python/InterpolateVolumeData.py

```diff
@@ -0,0 +1,372 @@
+
+        assert index == len(tensors[0])
+        target_vol.write_volume_data(obs,
+                                     source_vol.get_observation_value(obs),
+                                     volume_data)
+
+    source_file.close()
+    target_file.close()
+
+
+def _parse_args(sys_args):
+    """
+    defines and parses the command line arguments
+
+    Parameters
+    ----------
+    sys_args arguments passed via command line
+
+    Returns
+    -------
+    A dictionary of the parsed command line arguments
+    """
+    parser = argparse.ArgumentParser(
+        description=
+        "Interpolate the tensor components of h5 files to a desired grid")
+
+    parser.add_argument("source_files",
+                        action="extend",
+                        type=lambda path: glob.glob(path),
+                        nargs='+',
+                        help="The path to one or multiple source .h5 files. "
+                        "Accepts glob wildcards.")
+
+    parser.add_argument(
+        "--target_files",
+        type=str,
+        nargs='+',
+        help="the path to where the interpolated function is written. "
+        "The number of target files must match the number of source files."
+        " When no target file is specified, the interpolated data "
+        "is written to the source file in a new volume data.")
+
+    parser.add_argument(
+        "--tensor_components",
+        type=str,
+        nargs='*',
+        help="The names of the tensors that are to be interpolated. "
+        "Accepts regular expression. "
+        "If none are specified, all tensors are interpolated.")
+
+    parser.add_argument("--grid_extents",
+                        type=int,
+                        nargs='+',
+                        default=3,
+                        help="The extents of the target grid. "
+                        "Can be different for each dimension e.g. '3 5 4'")
+
+    parser.add_argument("--grid_basis",
+                        type=str,
+                        nargs='+',
+                        choices=basis_dict.keys(),
+                        default="Legendre",
+                        help="The basis of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument("--grid_quadrature",
+                        type=str,
+                        nargs='+',
+                        choices=quadrature_dict.keys(),
+                        default="GaussLobatto",
+                        help="The quadrature of the target grid. "
+                        "Can be different for each dimension")
+
+    parser.add_argument("--source_volume",
+                        type=str,
+                        nargs='+',
+                        default=["/VolumeData"],
+                        help="The name of the volume data object within the "
+                        "source file in which the data is contained")
+
+    parser.add_argument(
+        "--target_volume",
```

Comment:
I think this arg should also be required, and just a single name for all files. It shouldn't be surprising where the output data ends up, so I think it's best to just have the user specify the location explicitly.